plugins/azure-devops/src/alpha/plugin.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  createApiExtension,
+  createApiFactory,
+  createPageExtension,
+  createPlugin,
+  discoveryApiRef,
+  identityApiRef,
+} from '@backstage/frontend-plugin-api';
+import { azureDevOpsApiRef, AzureDevOpsClient } from '../api';
+import {
+  compatWrapper,
+  convertLegacyRouteRef,
+} from '@backstage/core-compat-api';
+import {
+  createEntityCardExtension,
+  createEntityContentExtension,
+} from '@backstage/plugin-catalog-react/alpha';
+import { azurePullRequestDashboardRouteRef } from '../routes';
+
+/** @alpha */
+export const azureDevOpsApi = createApiExtension({
+  factory: createApiFactory({
+    api: azureDevOpsApiRef,
+    deps: { discoveryApi: discoveryApiRef, identityApi: identityApiRef },
+    factory: ({ discoveryApi, identityApi }) =>
+      new AzureDevOpsClient({ discoveryApi, identityApi }),
+  }),
+});
+
+/** @alpha */
+export const azureDevOpsPullRequestPage = createPageExtension({
+  defaultPath: '/azure-pull-requests',
+  routeRef: convertLegacyRouteRef(azurePullRequestDashboardRouteRef),
+  loader: () =>
+    import('../components/PullRequestsPage').then(m =>
+      compatWrapper(<m.PullRequestsPage />),
+    ),
+});
+
+/** @alpha */
+export const entityAzurePipelinesContent = createEntityContentExtension({
+  name: 'pipelines',
+  defaultPath: '/pipelines',
+  defaultTitle: 'Pipelines',
+  loader: () =>
+    import('../components/EntityPageAzurePipelines').then(m =>
+      compatWrapper(<m.EntityPageAzurePipelines />),
+    ),
+});
+
+/** @alpha */
+export const entityAzureGitTagsContent = createEntityContentExtension({
```

Comment:
https://backstage.io/docs/frontend-system/architecture/naming-patterns#extensions

is the naming pattern instead `azureDevopsGitTagsEntityContent`? Same for the others. Note that we keep the namespace, kind, and name all collected instead of splitting them apart even if they are multi-word

Reply:
Fixed up 👍